### PR TITLE
Add test_axhline in test_datetime.py

### PR DESCRIPTION
**Pull request by CozyFrog**
## PR summary
I have added a datetime (no timedelta; this function does not support it, and I believe that is correct) smoke test for `Axes.axhline` in `lib/matplotlib/test/test_datetime.py`.
This addresses the `Axes.axhline` task from #26864.

The image below is the plot generated from my example code.
test_axhline

## PR checklist

- [N/A] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [N/A] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [N/A] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [N/A] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/coding_guide.html#new-features-and-api-changes)
- [N/A] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

File: lib/matplotlib/tests/test_datetime.py [hunk 1]
```diff
@@ -29,8 +29,20 @@ def test_arrow(self):
...
+                               datetime.datetime(2020, 5, 30),
+                               datetime.datetime(2020, 10, 12),
+                               datetime.datetime(2020, 11, 15)])
+        year_start = datetime.datetime(2020, 1, 1)
+        year_end = datetime.datetime(2020, 12, 31)
+        age = [21, 53, 20, 24]
+        ax.set_xlabel('Birth Date')
+        ax.set_ylabel('Age')
+        ax.set_ylim(bottom=year_start, top=year_end)
+        ax.bar(x=age, height=birth_date)
```
Comment (ksunden) [inline lib/matplotlib/tests/test_datetime.py:44]: I would say to keep this even more minimal and not have the bar graph (I think setting the ylim is important here)

If anything, I'd just do a `ax.plot`, but even that I think is not testing what we are looking to test here.

**Review by ksunden — APPROVED** [review]

File: lib/matplotlib/tests/test_datetime.py [hunk 2]
```diff
@@ -29,8 +29,9 @@ def test_arrow(self):
     @pytest.mark.xfail(reason="Test for axhline not written yet")
```
Comment (scottshambaugh) [inline lib/matplotlib/tests/test_datetime.py:29]: this line should be removed

**Review by scottshambaugh — COMMENTED** [review]
Could you add subplots to test different combinations of numbers and datetimes on different axes? Similar to the test you added here: https://github.com/matplotlib/matplotlib/pull/27084

**Comment by CozyFrog** [comment]
Sure thing. Can you elaborate on what you mean by different axes? I can test axhline on different plots, but I am unsure how to utilize it on different axes as it itself only plots on the horizontal. Below is an image of what I have come up with so far. Thank you for your feedback!
axhline_3figures

**Comment by scottshambaugh** [comment]
@CozyFrog ah I mean having subplots where the X/Y data are different combinations of datetime vs numbers. Right now you have datetimes on the Y axis with numbers on the X axis, and adding the other permutations of that should exercise these codepaths more fully (I think).

**Comment by ksunden** [comment]
@scottshambaugh I don't think that actually makes sense for this method, honestly... the x axis values are in `axes` coordinates, not data coordinates (well, unless you hand a `transform` argument), so really the only value handed to `axhline` that matters for units is the `y` value.

**Comment by scottshambaugh** [comment]
I'll defer to you on that @ksunden, I'm admittedly not super familiar with how the code paths for the coordinate transforms work work. If it's drawn directly in axes coordinates they I think you're right, my thought was if the xrange was cast to the data coordinates before being drawn.

**Comment by CozyFrog** [comment]
Thank you for the discussion @scottshambaugh & @ksunden – how should I move forward with this PR? Does this require any modifications?

**Review by scottshambaugh — APPROVED** [review]

**Pull request merged**